sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/action/ReflectiveActionRouter.scala

```diff
@@ -76,21 +87,31 @@ class ReflectiveActionRouter[A <: Action](
 
     val componentMethod = commandHandlerLookup(commandName)
 
-    val context =
-      InvocationContext(
-        message.payload().asInstanceOf[ScalaPbAny],
-        componentMethod.requestMessageDescriptor,
-        message.metadata())
-
-    val inputTypeUrl = message.payload().asInstanceOf[ScalaPbAny].typeUrl
-    componentMethod.lookupInvoker(inputTypeUrl) match {
-      case Some(methodInvoker) =>
-        val response = methodInvoker.invoke(action, context).asInstanceOf[Flux[Action.Effect[_]]]
-        Source.fromPublisher(response)
-      case None if ignoreUnknown => Source.empty()
-      case None =>
-        throw new NoSuchElementException(
-          s"Couldn't find any method with input type [$inputTypeUrl] in Action [$action].")
+    // lookup ComponentClient
+    val componentClients =
+      Reflect
+        .lookupComponentClientField(action.getClass, action)
```

Comment:
Small bug found when trying this with the sample..

```suggestion
        .lookupComponentClientField(action)
```

Reply:
I changed the signature from (Class, Any) to (Any), but how is it possible that the compiler didn't pick it?

Reply:
The other method receives `Any` and this was considered a `Tuple2`.. so it worked. I was just not sure if this was because of some implicit lying around or if it doesn't even need that.. 🤔

Reply:
Yeah, I was about to ask which error you saw. 
To be honest, I thought this kind of language quirk was already out of 2.13, but apparently not. 

Or we probably had a compiler warning, but didn't see because we have some many already.

By the way, I working on a branch that will make warnings fatal. Like we do in Akka and Kalix Runtime.